ingest-mongodb-public/src/sources/wiredTiger.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  HandleHtmlPageFuncOptions,
+  handleHtmlDocument,
+  makeGitDataSource,
+} from "mongodb-rag-ingest/sources";
+
+const removeElements = (domDoc: Document) => [
+  ...Array.from(domDoc.querySelectorAll("head")),
+  ...Array.from(domDoc.querySelectorAll("script")),
+  ...Array.from(domDoc.querySelectorAll("noscript")),
+  ...Array.from(domDoc.querySelectorAll("#top")),
+  ...Array.from(domDoc.querySelectorAll(".navpath")),
+];
+const extractTitle = (domDoc: Document) => {
+  const title = domDoc.querySelector("title");
+  if (title?.textContent?.includes("WiredTiger:")) {
+    return title?.textContent?.replace("WiredTiger: ", "").trim() ?? undefined;
+  }
+  return title?.textContent ?? undefined;
+};
+
+const extractMetadata = (domDoc: Document) => {
+  const version = domDoc.querySelector("#projectnumber")?.textContent;
+  return { version };
+};
+const htmlParserOptions: Omit<HandleHtmlPageFuncOptions, "sourceName"> = {
+  pathToPageUrl: (pathInRepo: string) =>
+    `https://source.wiredtiger.com${pathInRepo}`,
+  removeElements,
+  extractTitle,
+  extractMetadata,
+};
+
+export const wiredTigerSourceConstructor = async () => {
+  return await makeGitDataSource({
+    name: "wired-tiger",
+    repoUri: "https://github.com/wiredtiger/wiredtiger.github.com.git",
+    repoOptions: {
+      "--depth": 1,
+      "--branch": "master",
+    },
+    metadata: {
+      productName: "WiredTiger",
+      tags: ["docs", "storage-engine"],
+    },
+    filter: (path: string) =>
+      path.endsWith(".html") &&
+      path.includes("develop/") &&
```

Comment:
> Are we sure that there aren't paths like /1.2.3/develop/struct_w_t___w_e_i_r_d_n_a_m_e where this would match but the page is in an old version that we don't want to include?

i'm pretty sure there aren't, but it is more failsafe to do what you propose using startsWith. will refactor to use that.